viz/src/machines/viz.ts

```diff
@@ -1,79 +1,214 @@
+                  },
+                },
+                "Reached end of steps": {
+                  on: {
+                    "Reset steps": {
+                      actions: "Reset step index",
+                      target: "Playing steps",
+                    },
+                  },
+                },
+                Idle: {
+                  on: {
+                    "Enable playing execution steps": {
+                      target: "Playing steps",
+                    },
+                  },
+                },
+              },
               on: {
-                Play: {
-                  target: "Automatic playing on",
+                "Set automatic playing delay": {
+                  actions: "Assign automatic playing delay to context",
+                },
+                Load: {
+                  target: ".Idle",
                 },
               },
             },
-            "Automatic playing on": {
-              after: {
-                "automatic playing delay": {
-                  actions: "Increment step index",
-                  target: "Automatic playing on",
-                  internal: false,
+            "Managing machine and input execution": {
+              initial: "Idle",
+              states: {
+                Idle: {
+                  on: {
+                    Load: {
+                      target: "Executing machine and input",
+                    },
+                  },
+                },
+                "Executing machine and input": {
+                  entry: [
+                    "Cache input and machine code into context",
+                    "Reset step index",
+                  ],
+                  exit: "Exit loading state from submit button",
+                  type: "parallel",
+                  states: {
+                    "Making request to server": {
+                      initial: "Sending request",
+                      states: {
+                        "Sending request": {
+                          invoke: {
+                            src: "Execute machine and input on server",
+                            onDone: [
+                              {
+                                actions: "Assign machine execution to context",
+                                target: "Received response",
+                              },
+                            ],
+                            onError: [
+                              {
+                                target:
+                                  "#Visualizer.Application is ready.Managing machine and input execution.Failed to execute machine and input",
+                              },
+                            ],
+                          },
+                        },
+                        "Received response": {
+                          type: "final",
+                        },
+                      },
+                    },
+                    "Delaying loading state": {
```

Comment:
I don't really understand this delay target here, unless I'm mistaken the submit button already manages his own delaying ?

Reply:
The delay is made inside a child state of `Executing machine and input`. When the machine exits `Executing machine and input` state, the action `Exit loading state from submit button` is called, which sends an event to the machine that manages the load button to quit loading state. Thanks to this delay, the loading state is shown for at least one second. If the request takes more than one second, we do not wait more time and exit loading state right now.

Reply:
Ok I see !